cache

```diff
@@ -8,7 +8,7 @@ DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 (
     cd "$DIR"
 
-    for x in $(eval "echo $(nix-instantiate $NIXOPTS --eval --read-write-mode -E 'let this = (import ./. {}); in this.nixpkgs.stdenv.lib.concatStringsSep " " this.cacheTargetSystems')") ; do
-        nix-copy-closure --gzip --to "$1" $(nix-build $NIXOPTS -j 8 --no-out-link --indirect --add-root "$PWD/gc-roots/cache-$x.drv" -E "import ./shell.nix { system = \"$x\"; }")
+    for x in $(eval "echo $(nix-instantiate $NIXOPTS --eval --read-write-mode cache.nix)") ; do
+        nix-copy-closure --gzip --to "$1" $(nix-build $NIXOPTS -j 8 --no-out-link --indirect --add-root "$PWD/gc-roots/cache-$x.drv" ./shell.nix --arg system "\"$x\"")
```

Comment:
Wouldn't --argstr be better?  This could have escaping problems in the current form.